lib/web_ui/lib/src/engine/semantics/selectable.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/ui.dart' as ui;
+
+import 'label_and_value.dart';
+import 'semantics.dart';
+
+/// Renders semantics objects that have selected (true/false) states.
+///
+/// See also [ui.SemanticsFlag.isSelected]
+class SemanticSelectable extends SemanticRole {
+  SemanticSelectable(SemanticsObject semanticsObject) : super.withBasics(
+    SemanticRoleKind.selectable,
+    semanticsObject,
+    preferredLabelRepresentation: LabelRepresentation.ariaLabel,) {
+      addTappable();
+  }
+
+  @override
+  void update() {
+    super.update();
+
+    if (semanticsObject.isFlagsDirty) {
+      setAriaRole('tab');
```

Comment:
is this only applicable to tab? will probably need to check whether tab is the only widget in flutter that uses isSelected

Reply:
`aria-selectable` can be used by [multiple role types](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-selected#description). Perhaps this should be a `SemanticBehavior`, and then all roles that can be selected can mix it in. However, we need to know in advance which roles can possibly be selected in order to know to add the behavior. Or we could have a `hasSelectedState` flag (see suggestion below).